src/test/java/ua/com/juja/microservices/teams/slackbot/exceptions/ExceptionHandlerTest.java

```diff
@@ -101,6 +106,99 @@ public void handleTeamExchangeException() throws Exception {
+                "@slack5,@slack6,@slack7,@slack8");
+        ApiError apiError1 = new ApiError(
+                400, "TMF-F2-D3",
+                "Sorry, but the user already exists in team!",
+                "The reason of the exception is that user already in team",
+                messageWithUuids1,
+                Collections.emptyList()
+        );
+        ApiError apiError2 = new ApiError(
+                400, "TMF-F2-D3",
+                "Sorry, but the user already exists in team!",
+                "The reason of the exception is that user already in team",
+                messageWithUuids2,
+                Collections.emptyList()
+        );
+
+        TeamExchangeException exception1 = new TeamExchangeException(apiError1, new RuntimeException("exception1"));
+        TeamExchangeException exception2 = new TeamExchangeException(apiError2, new RuntimeException("exception2"));
+
+        when(teamService.activateTeam(activateTeamCommandText1)).thenThrow(exception1);
+        when(restTemplate.postForObject(eq(responseUrl1), any(RichMessage.class), eq(String.class))).thenReturn("");
+        when(userService.replaceUuidsBySlackNamesInExceptionMessage(messageWithUuids1)).thenReturn(messageWithSlackNames1);
+
+        when(teamService.activateTeam(activateTeamCommandText2)).then(invocation -> pauseThread(exception2));
+        when(restTemplate.postForObject(eq(responseUrl2), any(RichMessage.class), eq(String.class))).thenReturn("");
+        when(userService.replaceUuidsBySlackNamesInExceptionMessage(messageWithUuids2)).thenReturn(messageWithSlackNames2);
+
+        //when
+        Callable<Boolean> call = () -> {
+            try {
+                mvc.perform(MockMvcRequestBuilders.post(SlackUrlUtils.getUrlTemplate(teamsSlackbotActivateTeamUrl),
+                        SlackUrlUtils.getUriVars("slashCommandToken", "/teams-activate",
+                                activateTeamCommandText2, responseUrl2))
+                        .contentType(MediaType.APPLICATION_FORM_URLENCODED))
+                        .andExpect(status().isOk())
+                        .andExpect(content().string(ACTIVATE_TEAM_MESSAGE));
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+            return true;
+        };
+        FutureTask<Boolean> task = new FutureTask<>(call);
+        new Thread(task).start();
+        Thread.sleep(2000);
+
+        mvc.perform(MockMvcRequestBuilders.post(SlackUrlUtils.getUrlTemplate(teamsSlackbotActivateTeamUrl),
+                SlackUrlUtils.getUriVars("slashCommandToken", "/teams-activate",
+                        activateTeamCommandText1, responseUrl1))
+                .contentType(MediaType.APPLICATION_FORM_URLENCODED))
+                .andExpect(status().isOk())
+                .andExpect(content().string(ACTIVATE_TEAM_MESSAGE));
+
+        Boolean SecondThreadIsFinished = task.get();
```

Comment:
Variable SecondThreadIsFinished is never used. You can shorten this line to task.get().

Reply:
This variable need only for understanding what happens.
"SecondThreadIsFinished" means that both threads finished and it is a time to verify all method invokations. Not earlier

Reply:
OK. I agree with you.